app/scripts/lib/transaction/metrics.ts

```diff
@@ -1391,3 +1315,77 @@ function isInsufficientNativeBalance(
 
   return new Numeric(totalCost, 16).greaterThan(new Numeric(nativeBalance, 16));
 }
+
+function getMMSwapOrBridgeType(type: TransactionType): string | null {
+  if (type === TransactionType.swap) {
+    return 'mm_swap';
+  } else if (type === TransactionType.bridge) {
+    return 'mm_bridge';
+  }
+  return null;
+}
+
+/**
+ * Determines the transaction type for metrics and whether it's a contract interaction
+ *
+ * @param type - The transaction type
+ * @param originalType - The original transaction type (for retry transactions)
+ * @returns  Object containing final type and whether it's a contract interaction
+ */
+function determineTransactionTypeAndContractInteraction(
```

Comment:
This is much cleaner ❤️ 

In future, we could even benefit from a standard type like `TransactionPropertiesBuilder` and we could split all these properties into separate files, each with their own tests and domain.